src/resources/common.ts

```diff
@@ -483,6 +484,158 @@ export function handleFormulaMathML($: any) {
+          return {
+            ...acc,
+            curr: '',
+            lookingForEndDelimiter: '$$',
+          };
+        }
+
+        if (next.endsWith(lookingForEndDelimiter)) {
+          // end formula
+          return {
+            texts,
+            // add current formula to formulas, leaving out the end delimiter
+            formulas: [
+              next.slice(0, next.length - lookingForEndDelimiter.length),
+              ...formulas,
+            ],
+            curr: '',
+            lookingForEndDelimiter: undefined,
+          };
+        } else {
+          // continue collecting formula chars
+          return {
+            ...acc,
+            curr: next,
+          };
+        }
+      } else {
+        const foundDelimiter = checkForStartingDelimiter(next);
+
+        if (foundDelimiter) {
+          const [startDelimiter, endDelimiter] = foundDelimiter;
+
+          // begin formula
+          return {
+            // add current text to texts, leaving out the start delimiter
+            texts: [
+              next.slice(0, next.length - startDelimiter.length),
+              ...texts,
+            ],
+            formulas,
+            curr: '',
+            lookingForEndDelimiter: endDelimiter,
+          };
+        } else {
+          // continue collecting text chars
+          return {
+            ...acc,
+            curr: next,
+          };
+        }
+      }
+    },
+    { texts: [], formulas: [], curr: '', lookingForEndDelimiter: undefined }
+  );
+
+  if (parsed.texts.length === 0) {
+    // nothing was parsed
+    return false;
+  }
+
+  if (parsed.lookingForEndDelimiter) {
+    // parsing ended in the middle of a formula, therefore this cannot be processed
+    // as a valid mathjax expression
+    console.warn(
+      `MathJax parsing failed: Formula expression does not have a closing delimiter '${
+        parsed.lookingForEndDelimiter
+      }', at position ${text.length - parsed.curr.length}`
+    );
+
+    return false;
+  }
+
+  // zip together separate texts and formulas arrays into a single list. we can expect there
+  // to be a corresponding formula at every index of texts, except for the last one
+  const elementList = parsed.texts.flatMap((t, i) => [
+    t,
+    `<formula_inline src="${parsed.formulas[i]}" subtype="latex"></formula_inline>`,
+  ]);
+
+  // add any remaining text in curr
+  if (parsed.curr) {
+    elementList.push(parsed.curr);
+  }
+
+  return elementList;
+};
+
+// Traverses all text nodes in the given element using DFS.
+// Executes the given fn for each text node found.
+function traverseTextNodes(
+  el: cheerio.Element,
+  fn: (el: cheerio.Element) => void
+): void {
+  if (el.type === 'text') fn(el);
+
+  if (el.type === 'tag') {
+    return el.children.forEach((c: cheerio.Element) =>
+      traverseTextNodes(c, fn)
+    );
+  }
+}
+
+function handleInlineMathJax($: cheerio.Root) {
+  $('p, feedback').each((i, item) => {
```

Comment:
Hints can also be dynamically delivered, so this selector doesn't cover all cases that we need to fix this bug.  There are also a bunch of other cases that this misses for us to achieve the normalization of all latex and mathml inside formulas.  List items, table cells, description list terms and descrptions, etc.  It may make sense to handle this within the XML processing space.  There is a specific handler there that processes individual text nodes (regardless of whether they exist in p, li, td, etc).  That will give us 100% coverage.

Reply:
Refactored MathJAX identification/parsing to occur in XML to JSON conversion step